components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/ParAuthServiceImpl.java

```diff
@@ -18,81 +18,109 @@
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
 import org.wso2.carbon.identity.oauth.common.OAuth2ErrorCodes;
 import org.wso2.carbon.identity.oauth.common.OAuthConstants;
 import org.wso2.carbon.identity.oauth.par.common.ParConstants;
 import org.wso2.carbon.identity.oauth.par.dao.ParDAOFactory;
 import org.wso2.carbon.identity.oauth.par.dao.ParMgtDAO;
 import org.wso2.carbon.identity.oauth.par.exceptions.ParCoreException;
-import org.wso2.carbon.identity.oauth.par.model.ParAuthResponseData;
+import org.wso2.carbon.identity.oauth.par.model.ParAuthData;
 import org.wso2.carbon.identity.oauth.par.model.ParRequestDO;
 
 import java.util.Calendar;
 import java.util.Map;
 import java.util.TimeZone;
 import java.util.UUID;
 
+import javax.xml.namespace.QName;
 
 /**
- * Provides authentication services.
+ * Provides PAR services.
  */
 public class ParAuthServiceImpl implements ParAuthService {
 
+    private static final Log log = LogFactory.getLog(ParAuthService.class);
     ParMgtDAO parMgtDAO = ParDAOFactory.getInstance().getParAuthMgtDAO();
 
     @Override
-    public ParAuthResponseData generateParAuthResponse(Map<String, String> parameters) throws ParCoreException {
+    public ParAuthData handleParAuthRequest(Map<String, String> parameters) throws ParCoreException {
 
         String uuid = UUID.randomUUID().toString();
-        long expiry = ParConstants.EXPIRES_IN_DEFAULT_VALUE_IN_SEC;
 
-        ParAuthResponseData parAuthResponse = new ParAuthResponseData();
-        parAuthResponse.setReqUriUUID(uuid);
-        parAuthResponse.setExpiryTime(expiry);
+        ParAuthData parAuthResponse = new ParAuthData();
+        parAuthResponse.setrequestURIReference(uuid);
+        parAuthResponse.setExpiryTime(getExpiresInValue());
 
         persistParRequest(uuid, parameters, getScheduledExpiry(System.currentTimeMillis()));
 
         return parAuthResponse;
     }
 
-    private void persistParRequest(String uuid, Map<String, String> params, long scheduledExpiryTime)
+    private void persistParRequest(String uuid, Map<String, String> params, long expiresIn)
             throws ParCoreException {
 
-        parMgtDAO.persistParRequest(uuid, params.get(OAuthConstants.OAuth20Params.CLIENT_ID),
-                scheduledExpiryTime, params);
+        parMgtDAO.persistRequestData(uuid, params.get(OAuthConstants.OAuth20Params.CLIENT_ID),
+                expiresIn, params);
     }
 
+    @Override
     public Map<String, String> retrieveParams(String uuid, String clientId) throws ParCoreException {
 
-        ParRequestDO parRequestDO = parMgtDAO.getParRequest(uuid);
-        parMgtDAO.removeParRequest(uuid);
-        isRequestUriExpired(parRequestDO.getScheduledExpiryTime());
-        isClientIdValid(clientId, parRequestDO.getClientId());
+        ParRequestDO parRequestDO = parMgtDAO.getRequestData(uuid);
+        parMgtDAO.removeRequestData(uuid);
+        validateRequestURI(parRequestDO.getExpiresIn());
```

Comment:
even though we say validateRequestURI we only pass in the expiry time. Lets rename the method appropriately.